client/v3/balancer/balancer.go

```diff
@@ -247,7 +264,11 @@ func (bb *baseBalancer) HandleSubConnStateChange(sc balancer.SubConn, s grpcconn
 		bb.updatePicker()
 	}
 
-	bb.currentConn.UpdateBalancerState(bb.connectivityRecorder.GetCurrentState(), bb.picker)
+	state := balancer.State{
+		ConnectivityState: bb.connectivityRecorder.GetCurrentState(),
+		Picker:            bb.picker,
+	}
+	bb.currentConn.UpdateState(state)
```

Comment:
The `ClientConn` interface is what has changed here:

- https://pkg.go.dev/google.golang.org/grpc@v1.29.1/balancer#ClientConn
- https://pkg.go.dev/google.golang.org/grpc@v1.35.0/balancer#ClientConn